### PR TITLE
swr local storage cache

### PR DESCRIPTION
**Pull request by hyusap**
took a quick stab at optimization, seems to work

### CODE REVIEW

**Comment by bLopata** [comment]
great stuff! I was leaning towards indexedDB for improved indexing (faster querying), better async support and larger storage limits but I think @VVoruganti also was thinking localStorage. Happy to merge this in and get my local indexeddb changes in at a later date (as needed) though. Just want to pull it down and test locally before approving.

File: www/utils/swrCache.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    // Before unloading the app, we write back all the data into `localStorage`.
+    window.addEventListener('beforeunload', () => {
+      const appCache = JSON.stringify(Array.from(map.entries()));
+      localStorage.setItem('app-cache', appCache);
+    });
+    return map;
+  }
+
+  return new Map();
+}
```
Comment (VVoruganti) [inline www/utils/swrCache.ts:19]: Is there a way to invalidate the data on logout or trigger it manually, don't want to cache stuff in the case of switching accounts

Comment (VVoruganti) [reply www/utils/swrCache.ts:19]: Doing a quick test when I log out and log in with a different account it doesn't load the data from the previous account but it does keep it in the storage

image

**Comment by VVoruganti** [comment]
Comments addressed in #183

**Pull request closed without merging**